llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -37,6 +37,9 @@ static cl::opt<unsigned> SLPMaxVF(
 InstructionCost
 RISCVTTIImpl::getRISCVInstructionCost(ArrayRef<unsigned> OpCodes, MVT VT,
                                       TTI::TargetCostKind CostKind) {
+  // Check if the type is valid for all CostKind
+  if (!VT.isVector())
+    return InstructionCost::getInvalid();
```

Comment:
Where is this getting called from with a scalar MVT? I would have thought that after scalarization we wouldn't have any vector instructions that would still be calling into this

Reply:
It is from LoopUnrollPass along the way from TTI.getInstructionCost to TTI.getShuffleCost for
```
%splat.splat.i.i.i = shufflevector <2 x float> zeroinitializer, <2 x float> zeroinitializer, <2 x i32> zeroinitializer
```
The data type is `<2 x float>`, but becomes `float` after `getTypeLegalizationCost(Tp);` then enters getRISCVInstructionCost().
I'm just not sure how a vector type can be generated without a vector support.

If it is true, we might consider checking if the type is still vector after legalization at the beginning of TTI functions which accept vector types.

Reply:
Ah ok, thanks for clarifying. I guess there's nothing stopping a frontend from just emitting vector LLVM IR independently of the autovectorizers.

I'm not strongly opinionated about where we check that the legalized type is vector, I presume both will result in an invalid cost

Reply:
Yeah, in our case it's because we're compiling OpenCL or SYCL, with vector data types in the source.